allennlp/data/tokenizers/word_filter.py

```diff
@@ -33,37 +33,90 @@ def filter_words(self, words: List[Token]) -> List[Token]:
-                              "'", '"', '&', '$', '#', '@', '(', ')', '?'])
+    def __init__(self,
+                 stopword_file: str=None,
+                 tokens_to_add: List[str]=None) -> None:
+        self._tokens_to_add = tokens_to_add or []
+        if stopword_file is not None:
+            self.stopwords = self._set_from_file(stopword_file)
+        else:
+            self.stopwords = set(['I', 'a', 'aboard', 'about', 'above', 'accordance', 'according',
+                                  'across', 'after', 'against', 'along', 'alongside', 'also', 'am',
+                                  'amid', 'amidst', 'an', 'and', 'apart', 'are', 'around', 'as',
+                                  'aside', 'astride', 'at', 'atop', 'back', 'be', 'because', 'before',
+                                  'behind', 'below', 'beneath', 'beside', 'besides', 'between',
+                                  'beyond', 'but', 'by', 'concerning', 'do', 'down', 'due', 'during',
+                                  'either', 'except', 'exclusive', 'false', 'for', 'from', 'happen',
+                                  'he', 'her', 'hers', 'herself', 'him', 'himself', 'his', 'how',
+                                  'how many', 'how much', 'i', 'if', 'in', 'including', 'inside',
+                                  'instead', 'into', 'irrespective', 'is', 'it', 'its', 'itself',
+                                  'less', 'me', 'mine', 'minus', 'my', 'myself', 'neither', 'next',
+                                  'not', 'occur', 'of', 'off', 'on', 'onto', 'opposite', 'or', 'our',
+                                  'ours', 'ourselves', 'out', 'out of', 'outside', 'over', 'owing',
+                                  'per', 'prepatory', 'previous', 'prior', 'pursuant', 'regarding',
+                                  's', 'sans', 'she', 'subsequent', 'such', 'than', 'thanks', 'that',
+                                  'the', 'their', 'theirs', 'them', 'themselves', 'then', 'these',
+                                  'they', 'this', 'those', 'through', 'throughout', 'thru', 'till',
+                                  'to', 'together', 'top', 'toward', 'towards', 'true', 'under',
+                                  'underneath', 'unlike', 'until', 'up', 'upon', 'us', 'using',
+                                  'versus', 'via', 'was', 'we', 'were', 'what', 'when', 'where',
+                                  'which', 'who', 'why', 'will', 'with', 'within', 'without', 'you',
+                                  'your', 'yours', 'yourself', 'yourselves', ",", '.', ':', '!', ';',
+                                  "'", '"', '&', '$', '#', '@', '(', ')', '?'])
+        for token in self._tokens_to_add:
+            self.stopwords.add(token)
+
+    def _set_from_file(self, stopword_file: str) -> Set[str]:
+        '''
+        Extract stopwords from a file.
+        Expected file format is one stopword per line.
+
+        Returns
+        _______
```

Comment:
These are underscores where they should be dashes.

For future reference, though, private methods like this with good, obvious names don't require such detailed docstrings (no harm in having them, but no need to go out of your way to add them, either).

Reply:
Except, looking at this name again, I'd expect a method called `self._set_something` to have no return value and to modify the internal state of `self`.  This does neither - it reads the lines from a file and returns a set, without modifying `self`.  That's also why pylint is complaining about this line, saying that there's no use of self.

This method feels like it should be called `read_set_from_file`, and live in `common.file_utils`, not in here.